community/modules/compute/gke-job-template/templates/gke-job-base.yaml.tftpl

```diff
@@ -25,13 +25,15 @@ spec:
                 - ${node_pool}
                 %{~ endfor ~}
       %{~ endif ~}
+      %{~ if has_node_selector ~}
```

Comment:
I suggest you take the 1 minute to test whether you can remove the extra local variable (which reproduced information you already need anyhow):

```hcl
%{~ if machine_family != null || length(node_selectors) > 0 ~}
```

Reply:
I find it less clear while reading the template since it (outwardly) appears to contain independent information. I won't block the PR on it, though, so effectively this may be a `nit`.

Reply:
I have refactored to not pass dup information while still keeping logic in the template to a minimum.

This was time well spent as I also found that `nodeSelectors` were generally broken.